rllib/core/optim/tests/test_marl_optimizer_torch.py

```diff
@@ -0,0 +1,193 @@
+        # 1.03 the return of each policy gets to around 100.
+        self.assertLess(loss["total_loss"], 1.03)
+
+    def test_marl_optimizer_set_state_get_state_torch(self):
+        torch.manual_seed(1)
+        env = gym.make("CartPole-v1")
+
+        config = {
+            "modules": {
+                "module_0": {
+                    "module_class": DiscreteBCTorchModule,
+                    "observation_space": env.observation_space,
+                    "action_space": env.action_space,
+                    "model_config": {"hidden_dim": 32},
+                },
+                "module_1": {
+                    "module_class": DiscreteBCTorchModule,
+                    "observation_space": env.observation_space,
+                    "action_space": env.action_space,
+                    "model_config": {"hidden_dim": 32},
+                },
+            }
+        }
+
+        marl_module1 = MultiAgentRLModule.from_multi_agent_config(config)
+        marl_optimizer1 = MultiAgentRLOptimizer.from_marl_module(
+            marl_module1,
+            rl_optimizer_classes={
+                "module_0": BCTorchOptimizer,
+                "module_1": BCTorchOptimizer,
+            },
+            optim_kwargs={"module_0": {"lr": 0.1}, "module_1": {"lr": 0.1}},
+        )
+
+        marl_module2 = MultiAgentRLModule.from_multi_agent_config(config)
+        marl_optimizer2 = MultiAgentRLOptimizer.from_marl_module(
+            marl_module2,
+            rl_optimizer_classes={
+                "module_0": BCTorchOptimizer,
+                "module_1": BCTorchOptimizer,
+            },
+            optim_kwargs={"module_0": {"lr": 0.2}, "module_1": {"lr": 0.2}},
+        )
+        self.assertIsInstance(marl_optimizer1.get_state(), dict)
+
+        # check that the learning rates are different before setting the state to be
+        # the same
+        check(
+            marl_optimizer1.get_state()["module_0"]["module"]["param_groups"][0]["lr"],
+            marl_optimizer2.get_state()["module_0"]["module"]["param_groups"][0]["lr"],
+            false=True,
+        )
+        check(
+            marl_optimizer1.get_state()["module_1"]["module"]["param_groups"][0]["lr"],
+            marl_optimizer2.get_state()["module_1"]["module"]["param_groups"][0]["lr"],
+            false=True,
+        )
+
+        marl_optimizer2.set_state(marl_optimizer1.get_state())
+        # check that the learning rates are the same after setting the state to be
+        # the same
+        check(
+            marl_optimizer1.get_state()["module_0"]["module"]["param_groups"][0]["lr"],
+            marl_optimizer2.get_state()["module_0"]["module"]["param_groups"][0]["lr"],
+        )
+        check(
+            marl_optimizer1.get_state()["module_1"]["module"]["param_groups"][0]["lr"],
+            marl_optimizer2.get_state()["module_1"]["module"]["param_groups"][0]["lr"],
+        )
+
+    def test_marl_optimizer_shared_encoder_torch(self):
```

Comment:
this is the experimental test that involves a new marl module that you were saying that we don't want to merge @kouroshHakha